api/org/commcare/api/persistence/SqlHelper.java

```diff
@@ -79,13 +79,14 @@ public static PreparedStatement prepareTableSelectStatement(Connection c,
                                                                 String storageKey,
                                                                 String[] fields,
                                                                 String[] values,
-                                                                Persistable p) {
+                                                                Persistable p) throws IllegalArgumentException{
```

Comment:
I'm not sure I agree with that, actually. I think it's fine to throw typed exceptions and document in the method signature when they are thrown so people can plan around using them semantically.

In general non-runtime exceptions are a better fit for that, but if we are using the exceptions semantically in any way, I think documenting them in the signature is better than not.